File: CCC-school-work/comporg/pnc2/ahought2/python/primeregbs.py
#!/usr/bin/python
import sys
import time

# printf function top STDERR
def printferr(format, *args):
    sys.stderr.write(format % args)
# printf function to STDOUT
def printfout(format, *args):
    sys.stdout.write(format % args)

# Newtons square root
def isqrt(n):
    x = n
    y = (x + 1) // 2
    while y < x:
        x = y
        y = (x + n // x) // 2
    return x

# make integer of first command line argument
if len(sys.argv) < 2:
    print("Error: incorrect number of arguments!")
    sys.exit(1)
else:
    qty = int(sys.argv[1])

start_time = time.time()
    
count = 0
number = 1 
check = 1

while count < qty:
    number += 1
    factor = 2 
    check = 1
    sqrt = isqrt(number)
    while factor <= sqrt:
        mod = number%factor
        if mod == 0:
            check = 0
            break
        factor += 1
    if check == 1:
        printferr("%d ", number)
        count += 1
printferr("\n")
printfout("%8.4lf\n", time.time() - start_time)

File: CCC-school-work/comporg/pnc2/ahought2/perl/primeregbs.pl
#!/usr/bin/perl

use strict;
use warnings;
use diagnostics;
use Time::HiRes qw( time );

#use feature 'say';
#use feature "switch";

# variables
my $number  = 0;
my $factor  = 0;
my $count   = 0;
my $check   = 0;
my $qty     = 0;
my $sqrt    = 0;

#Error checking
if (!defined $ARGV[0]) # If no argument
{
    print "ERROR: missing quantity!\n";
    exit 1;
}
else
{
    $qty = $ARGV[0];    # set qty to command line argument
    if ($qty < 0)
    {
        print "ERROR: must choose a positive integer!\n";
        exit 2;
    }
}

my $startTime = [Time::HiRes::gettimeofday()];

for ($number = 2; $count < $qty; $number++) #iterate through numbers to check
{
    $check = 1;
    $sqrt = sqrt($number);
    for ($factor = 2; $factor <= $sqrt; $factor++)  #iterate through factors to check numbers against
    {
        if (($number % $factor) == 0)
        {
            $check = 0; # proven to NOT be prime
            last;
        }
    }

    if ($check == 1)    # if check is still true it was Not diproved as a Prime
    {
        print STDERR "$number ";    # print to stderr
        $count++;
    }
}

print "\n";

#my $endTime = [Time::HiRes::gettimeofday()];
my $endTime = Time::HiRes::tv_interval($startTime);
#print STDOUT "$endTime\n";
printf STDOUT ("%8.4lf\n", $endTime);
